implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicMeterRegistry.java

```diff
@@ -194,18 +208,19 @@ protected void publish() {
     // VisibleForTesting
     Stream<String> writeMeter(Meter meter) {
         // Snapshot values should be used throughout this method as there are chances for values to be changed in-between.
-        List<Attribute> attributes = new ArrayList<>();
+        Map<String, Attribute> attributes = new HashMap<>();
         for (Measurement measurement : meter.measure()) {
             double value = measurement.getValue();
             if (!Double.isFinite(value)) {
                 continue;
             }
-            attributes.add(new Attribute(measurement.getStatistic().getTagValueRepresentation(), value));
+            String name = measurement.getStatistic().getTagValueRepresentation();
+            attributes.put(name, new Attribute(name, value));
```

Comment:
This results in a last duplicate attribute name wins behavior. I'm wondering what was the effective behavior before when we were sending all to New Relic? Was it resulting in an error or only saving one of the attributes with the duplicate name, or something else?

Reply:
I'll investigate. If you haven't had issues, and error isn't returned by the client and/or non-deterministic behavior occurs. I came across this while working on my Java-Agent based registry PR.  The Agent Insights API takes a Map and forwards to an API. I assume the same or similar. When reading the docs, it made sense to align to a similar client side Map behavior for consistency.  @breedx-nr would you know?